### PR TITLE
chore: add .gitignore

### PR DESCRIPTION
**Pull request by aaravm**
Adding the gitignore file. It contains:

- target/ and debug/ : Common Practice
- cargo.lock: Since this is a library, we don't need this file
- openapitools.json: leftover from autogenerated models from build-models.sh
- funnel: From funnel, being used for tests

### CODE REVIEW

**Review by pavelnikonorov — APPROVED** [review]
LGTM!

File: .gitignore
Comment (uniqueg) [inline .gitignore:1]: Consider using auto-generated ignore patterns as a basis for this file, e.g., from here: https://www.toptal.com/developers/gitignore/api/rust

Comment (aaravm) [reply .gitignore:1]: I think I have updated according to the link, could you please see it again, and suggest whether I should I change it? @uniqueg

Comment (uniqueg) [reply .gitignore:1]: I was more thinking of copying it over including the comments, then add a section at the top for custom additions. But given how small the file is (compared to, say, a comprehensive list of ignore patterns for Python), I don't think we need to go through that trouble, at least at this point.

**Review by uniqueg — APPROVED** [review]

**Review by uniqueg — APPROVED** [review]

**Pull request merged**